app/models/maintenance_tasks/run.rb

```diff
@@ -443,6 +443,20 @@ def masked_arguments
       argument_filter.filter(arguments)
     end
 
+    # Appends output to the existing output in the database if the column exists.
+    #
+    # @param new_output [String] the output to append.
+    def append_output(new_output)
+      return unless self.class.column_names.include?("output")
+
+      # Reload output from database to get the latest value
+      current_output = self.class.where(id: id).pluck(:output).first || ""
```

Comment:
Unfortunately I don't think we can afford adding a feature like that to the framework. It may work okay for smaller applications but for large applications with a lot of rows to iterate over the approach of transferring large chunks for data back-and-forth doesn't scale well